### PR TITLE
Add taxonomy tags to organize content

### PR DESCRIPTION
**Pull request by Reshrahim**
Thank you for helping make the Radius documentation better!

**Please follow this checklist before submitting:**

- [x] [Read the contribution guide](https://docs.radapp.dev/contributing/docs/)
- [x] Commands include options for Linux, MacOS, and Windows within codetabs
- [x] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [x] Images use HTML style and have alternative text
- [x] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description
This PR leverages the concept of taxonomies in Hugo. Created a shortcode to categorize content on a page by tags taxonomy. Testing the shortcode with quickstart

image

Separate PRs to follow on adding categories to organize by type of content (quickstarts vs how-tos) and also by both categories and tags 

This will be useful with reference apps/samples as there is an influx of the content 

## Issue reference
#515

### CODE REVIEW

File: docs/content/author-apps/dapr-resource/dapr-buildingblocks.png
Comment (AaronCrawfis) [inline docs/content/author-apps/dapr-resource/dapr-buildingblocks.png:1]: Are all the Dapr directories meant to be renamed?

File: docs/layouts/shortcodes/categorize.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    &nbsp;
+      <div class="card">
+        <div class="card-body">
+          <h3 class="card-title"><b>{{ $key }}</b></h3>
+          {{ range $taxonomy.Pages }}
+          <p class="card-text"><li hugo-nav="{{ .RelPermalink}}"><a href="{{ .Permalink}}">{{ .Title }}</a></li></p>
+          {{end}}
+        </div> 
+      </div>
+      {{end}}
```
Comment (AaronCrawfis) [inline docs/layouts/shortcodes/categorize.html]: There's a missing `</div>` and the end here

File: docs/content/getting-started/quickstarts/_index.md [hunk 2]
```diff
@@ -4,5 +4,7 @@ title: "Quickstart guides"
 linkTitle: "Quickstarts"
 description: "Learn about Project Radius topics via quickstart guides, complete with code samples"
 weight: 200
-card_list: true
+no_list: true
 ---
+
+{{<categorize>}}
```
Comment (AaronCrawfis) [inline docs/content/getting-started/quickstarts/_index.md]:
```suggestion
{{< categorize >}}
```

Also this is AWESOME!!

**Review by AaronCrawfis — CHANGES_REQUESTED** [review]
This is AWESOME!!

On the term "categories", I'm wondering if we can make sure we're using the right term.

A category to me is one of:
  - Quickstart
  - How-To Guide
  - Concept

where the category describes what kind of document this is.

I think "Topic" might be better, which describes what topic is within the document.

**Comment by Reshrahim** [comment]
> This is AWESOME!!
> 
> On the term "categories", I'm wondering if we can make sure we're using the right term.
> 
> A category to me is one of:
> 
> * Quickstart
> * How-To Guide
> * Concept
> 
> where the category describes what kind of document this is.
> 
> I think "Topic" might be better, which describes what topic is within the document.

Yeah, I just experimented with categories but we can definitely adjust them to our needs. Categories and Tags are available as default taxonomies and we can also add custom taxonomy

So category can represent the type of content (concept/how-to/quickstart guide) whereas tags/topics can represent the keywords related to content 

We can also make the shortcode organize content either by category or tags based on the level eg: Getting started section / Quickstarts

**Comment by jasonviviano** [comment]
Hey, do we have to edit anything in the `config.toml`? I was looking at the documentation website and for a customized `series` tag we'd have to add in something like below, not sure if this requires something similar?

```
taxonomies:
  category: categories
  series: series
  tag: tags
 ```

Currently the `config.toml` only has:

```
# Taxonomies
[taxonomies]
category = "categories"
```

So not sure if tags would also work correctly right now.

**Comment by Reshrahim** [comment]
> Hey, do we have to edit anything in the `config.toml`? I was looking at the documentation website and for a customized `series` tag we'd have to add in something like below, not sure if this requires something similar?
> 
> ```
> taxonomies:
>   category: categories
>   series: series
>   tag: tags
> ```
> 
> Currently the `config.toml` only has:
> 
> ```
> # Taxonomies
> [taxonomies]
> category = "categories"
> ```
> 
> So not sure if tags would also work correctly right now.

so defaults are categories and tags which would work without this config in `config.toml` too. If you want to add custom taxonomy terms like series/topic we need to put it in `config.toml` like above

File: docs/content/author-apps/dapr-resource/dapr-buildingblocks.png
Comment (Reshrahim) [reply docs/content/author-apps/dapr-resource/dapr-buildingblocks.png:1]: yup it wasn't allowing the category to be `dapr` as the content folder under developer guides also reads the same as the URL generation for the site conflicts

Comment (Reshrahim) [reply docs/content/author-apps/dapr-resource/dapr-buildingblocks.png:1]: but the URL generated has `categories/dapr` which shouldn't conflict. let me check further

File: docs/layouts/shortcodes/categorize.html (hunk 1 above)
Comment (Reshrahim) [reply docs/layouts/shortcodes/categorize.html]: updated

File: docs/content/author-apps/dapr-resource/dapr-buildingblocks.png
Comment (Reshrahim) [reply docs/content/author-apps/dapr-resource/dapr-buildingblocks.png:1]: Looks like this is a known limitation from Hugo. Content section cannot be the same as the taxonomy. So we either use a different name for dapr content section or we use different taxonomy for dapr related pages

File: docs/layouts/shortcodes/categorize.html (hunk 1 above)
Comment (AaronCrawfis) [reply docs/layouts/shortcodes/categorize.html]: I'm still seeing a missing a </div> that doesn't close the card-pane div from line 1

Comment (Reshrahim) [reply docs/layouts/shortcodes/categorize.html]: pushing the change now! missed to commit

File: docs/content/getting-started/quickstarts/quickstart-dapr/_index.md [hunk 3]
```diff
@@ -6,14 +6,15 @@ description: "Learn Project Radius by authoring templates and deploying a Dapr a
 weight: 300
 no_list: true
 slug: "dapr"
+tags : ["dapr"]
```
Comment (AaronCrawfis) [inline docs/content/getting-started/quickstarts/quickstart-dapr/_index.md]: Could we use proper casing for brands?

```suggestion
tags : ["Dapr"]
```

File: docs/content/getting-started/quickstarts/quickstart-container-wi/index.md [hunk 4]
```diff
@@ -5,6 +5,7 @@ linkTitle: "Connect to Azure resources"
 description: "Learn how to connect a container to an Azure resource with managed identities and RBAC" 
 weight: 600
 slug: 'azure-connection'
+tags: ["azure","containers"]
```
Comment (AaronCrawfis) [inline docs/content/getting-started/quickstarts/quickstart-container-wi/index.md]:
```suggestion
tags: ["Azure","containers"]
```

File: docs/content/getting-started/quickstarts/quickstart-aws/index.md [hunk 5]
```diff
@@ -4,7 +4,7 @@ title: "Quickstart: Deploy AWS resources"
 linkTitle: "Deploy AWS resources"
 description: "Learn about how to setup an AWS cloud provider and deploy an AWS MemoryDB for Redis with Radius"
 weight: 500
-slug: "aws"
+tags: ["aws"]
```
Comment (AaronCrawfis) [inline docs/content/getting-started/quickstarts/quickstart-aws/index.md]:
```suggestion
tags: ["AWS"]
```

File: docs/content/getting-started/quickstarts/quickstart-keyvault-wi/index.md [hunk 6]
```diff
@@ -5,6 +5,7 @@ linkTitle: "Persistent Key Vault volume"
 description: "Learn how to mount an Azure Key Vault as a volume to a container" 
 weight: 700
 slug: 'volume-keyvault'
+tags: ["azure","containers"]
```
Comment (AaronCrawfis) [inline docs/content/getting-started/quickstarts/quickstart-keyvault-wi/index.md]:
```suggestion
tags: ["Azure","containers"]
```

File: docs/content/author-apps/dapr-resource/dapr-buildingblocks.png
Comment (AaronCrawfis) [inline docs/content/author-apps/dapr-resource/dapr-buildingblocks.png:1]: Could we rename this section to `dapr-resources` and add a slug for `dapr` please?

**Review by AaronCrawfis — CHANGES_REQUESTED** [review]
Looking good!

Added some casing notes and a note on the Dapr resources section.

Could you also add a file to `content/tags` with frontmatter? The tags page currently doesn't have a sidebar on the left side of the page, so a user could get lost. Adding a page with just frontmatter will cause the sidebar to be loaded.

**Comment by Reshrahim** [comment]
> Looking good!
> 
> Added some casing notes and a note on the Dapr resources section.
> 
> Could you also add a file to `content/tags` with frontmatter? The tags page currently doesn't have a sidebar on the left side of the page, so a user could get lost. Adding a page with just frontmatter will cause the sidebar to be loaded.

something like this ?
image

File: docs/content/_index.md [hunk 7]
```diff
@@ -48,7 +48,7 @@ Teams can offload the complexity of wiring-up applications and let Radius employ
       <a href="{{< ref networking >}}"><img src="network-logo.svg" alt="Networking logo" style="width:40%"></a>
     </td>
     <td style="width:25%;text-align:center">
-      <a href="{{< ref dapr >}}"><img src="dapr-logo.svg" alt="Dapr logo" style="width:40%"></a>
+      <a href="{{< ref dapr-resources >}}"><img src="dapr-logo.svg" alt="Dapr logo" style="width:40%"></a>
```
Comment (AaronCrawfis) [inline docs/content/_index.md:51]: could we do a search and make sure there aren't any other {{< ref dapr >}} we need to change?

Comment (Reshrahim) [reply docs/content/_index.md:51]: yes did a quick pass..no other dapr references

File: docs/content/author-apps/dapr-resources/index.md
Comment (AaronCrawfis) [inline docs/content/author-apps/dapr-resources/index.md:1]: Could we try adding a slug for `dapr` on this page?

**Review by AaronCrawfis — APPROVED** [review]

**Pull request merged**